Classes/Extensions/Amazing+Tools.swift

```diff
@@ -9,198 +9,184 @@
 import Foundation
 import CoreGraphics
 
-
 public extension Amazing {
     
+    /// Returns an image with the selected icon
+    ///
+    /// - Parameters:
+    ///   - size: the size of the returned image
+    ///   - color: the foreground color of the icon
+    ///   - backgroundColor: the background color of the icon
+    /// - Returns: an image that contains the selected icon
     public func asImage(size: CGSize, color: Color = Color.black, backgroundColor: Color = Color.clear) -> Image {
         return Image(icon: self, size: size, color: color, backgroundColor: backgroundColor)
     }
     
+    /// Returns a square image with the selected icon
+    ///
+    /// - Parameters:
+    ///   - size: the height and width of the returned image
+    ///   - color: the foreground color of the icon
+    ///   - backgroundColor: the background color of the icon
+    /// - Returns: a square image that contains the selected icon
     public func asImage(size: CGFloat, color: Color = Color.black, backgroundColor: Color = Color.clear) -> Image {
         return Image(icon: self, size: CGSize(width: size, height: size), color: color, backgroundColor: backgroundColor)
     }
     
+    /// Returns an attributed string with the selected icon
+    ///
+    /// - Parameters:
+    ///   - fontSize: the size of the font used in the string
+    ///   - color: the foreground color of the selected icon
+    ///   - backgroundColor: the background color of the selected icon
+    /// - Returns: an attributed string that contains the selected icon
     public func asAttributedText(fontSize: CGFloat, color: Color = Color.black, backgroundColor: Color = Color.clear) -> NSAttributedString {
-        return NSAttributedString.awesome(icon: self, fontSize: fontSize, color: color, backgroundColor: backgroundColor)
+        return NSAttributedString(icon: self, fontSize: fontSize, color: color, backgroundColor: backgroundColor)
     }
     
 }
 
-
-extension Awesome.brand {
-
-    public typealias AwesomeType = Awesome.brand
-
-    public static func fromType(_ code: String) -> AwesomeType? {
-        guard let icon = AwesomeType(rawValue: code) else {
-            return nil
-        }
-        return icon
-    }
-
-    public var name: String {
-        let index = AwesomeType.all.index(of: self)!
-        let name = AwesomeType.keys[index]
-        return name
+extension Amazing {
+    
+    public var detailedKey: String {
```

Comment:
These still missing comments on what they do?

Reply:
They were already in declaration of the `Amazing` protocol and I thought this would be enough. But it looks like the properties appear twice in autocomplete: once with (protocol) and once without (extension) the hint. I'll add them to the extension as well.

Reply:
Looks like the same thing applies for the deprecation warnings. So I'll add them to the protocol again too.